plugin/web/extensions/sidebar/popups/buildingSearch/types.ts

```diff
@@ -1,22 +1,65 @@
-export type IndexData = {
-  field: string;
-  values: string[];
+export type InitData = {
+  viewport: Viewport;
+  data: RawDatasetData;
+};
+
+export type RawDatasetData = {
+  title: string;
+  dataID: string;
+  searchIndex: {
+    url: string;
+  }[];
+};
+
+export type SearchIndex = {
+  baseURL: string;
+  indexRoot: {
+    indexes: {
+      [key: string]: {
+        kind: string;
+        values: {
+          [key: string]: {
+            count: number;
+            url: string;
+          };
+        };
+      };
+    };
+  };
+  resultsData?: any[];
 };
```

Comment:
Can you create a separate type for indexes?
It'll be a bit cleaner and we might want to support other type of indexes soon.

Ref:
<img width="560" alt="Screenshot 2023-02-20 at 14 28 21" src="https://user-images.githubusercontent.com/42397980/220059915-de392343-9189-4621-9d09-76fc133a6893.png">

Reply:
Okey sure!